File: src/main/java/io/spring/springbatch/CustomBatchConfigurer.java
package io.spring.springbatch;

import org.springframework.batch.core.repository.JobRepository;
import org.springframework.batch.core.repository.support.JobRepositoryFactoryBean;
import org.springframework.boot.autoconfigure.batch.BasicBatchConfigurer;
import org.springframework.boot.autoconfigure.batch.BatchProperties;
import org.springframework.boot.autoconfigure.transaction.TransactionManagerCustomizers;
import org.springframework.context.annotation.Configuration;
import javax.sql.DataSource;

@Configuration
public class CustomBatchConfigurer extends BasicBatchConfigurer {
    private final DataSource dataSource;

    /**
     * Create a new {@link BasicBatchConfigurer} instance.
     *
     * @param properties                    the batch properties
     * @param dataSource                    the underlying data source
     * @param transactionManagerCustomizers transaction manager customizers (or
     *                                      {@code null})
     */
    protected CustomBatchConfigurer(final BatchProperties properties, final DataSource dataSource, final TransactionManagerCustomizers transactionManagerCustomizers) {
        super(properties, dataSource, transactionManagerCustomizers);
        this.dataSource = dataSource;
    }

    @Override
    protected JobRepository createJobRepository() throws Exception {
        final JobRepositoryFactoryBean factory = new JobRepositoryFactoryBean();
        factory.setDataSource(this.dataSource);
        factory.setTransactionManager(this.getTransactionManager());
        factory.setIsolationLevelForCreate("ISOLATION_READ_COMMITTED");
        factory.setTablePrefix("SYSTEM_");
        return factory.getObject();
    }
}

File: src/main/kotlin/com/spring/batch/CustomBatchConfigurer.kt
package com.spring.batch

import org.springframework.batch.core.DefaultJobKeyGenerator
import org.springframework.batch.core.configuration.annotation.EnableBatchProcessing
import org.springframework.batch.core.repository.JobRepository
import org.springframework.batch.core.repository.support.JobRepositoryFactoryBean
import org.springframework.batch.item.database.support.DefaultDataFieldMaxValueIncrementerFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.transaction.PlatformTransactionManager
import org.springframework.transaction.annotation.Isolation
import javax.sql.DataSource


//@Configuration
//@EnableBatchProcessing(
//    daft
//)
//class CustomBatchConfigurer(
//    private val dataSource: DataSource,
//    private val platformTransactionManager: PlatformTransactionManager,
//) {
//    @Bean
//    fun createJobRepository(): JobRepository {
//        val factory = JobRepositoryFactoryBean()
//        factory.setDatabaseType("POSTGRES")
//        factory.setDataSource(dataSource)
//        factory.setIncrementerFactory(DefaultDataFieldMaxValueIncrementerFactory(dataSource))
//        factory.setJobKeyGenerator(DefaultJobKeyGenerator())
//        factory.transactionManager = this.platformTransactionManager
//        factory.setIsolationLevelForCreateEnum(Isolation.READ_COMMITTED)
//        factory.setTablePrefix("SYSTEM_")
//        return factory.getObject()
//    }
//}
